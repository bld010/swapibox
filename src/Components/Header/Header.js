import React from 'react';
import './Header.scss';
import PropTypes from 'prop-types';
import TextCrawl from '../TextCrawl/TextCrawl'
import FilterButton from '../FilterButton/FilterButton'

function Header(props) {



  return(
    <header className="Header">
      <h1>SWAPIbox</h1>
      <TextCrawl movie={props.movie}/>
      <nav>
        <FilterButton filterType="people"/>
        <FilterButton filterType="planets"/>
        <FilterButton filterType="vehicles" />
        <FilterButton filterType="favorites" favoritesCount={props.favoritesCount} />  
      </nav>
    </header>
  )
}


export default Header;

Header.propTypes = {
  favoritesCount: PropTypes.number,
  movie: PropTypes.object,
}