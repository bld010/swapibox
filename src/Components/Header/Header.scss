$main-color: green;
$button-hover-color: lightgreen;
$title-color: goldenrod;

.Header {
  background: black;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  overflow: hidden;
  background-image: url('../../starBackground.jpg');


  & .placeholder {
    min-width: 400px;
    min-height: 200px;

    & h4 {
      color: lightblue;
      font-size: 20px;
      margin-top: 130px;
    }
  }

  & .active {
    color: $title-color;
    text-decoration: none;
  }

  & a {
    text-decoration: none;
    color: $main-color;
    font-family: 'Courier';
    font-size: 30px;
    text-align: left;
    font-weight: 300;
  }

  & h1 {
    margin: 100px 0px 10px 0px;
    padding: 10px;
    font-size: 50px;
    color: $title-color;
  }

  & nav {
    margin: -15px 0px 20px 0px;
    padding: 120px 10px 50px 20px;
    display: flex;
    flex-direction: column;
    width: 250px;
    border: 2px solid $main-color;
    background: black;
    box-shadow: 0px 0px 8px 3px rgb(67, 165, 71);
    border-radius: 5px;
    

    & button {
      margin: 5px 25px;
      background: none;
      border: none;
      font-size: 25px;
      line-height: 0.06;
      padding: 0;
      text-align: left;
      color: $main-color;

      & :hover {
        cursor: pointer;
        color: $button-hover-color;
        
      }
      
    }
  }
}

