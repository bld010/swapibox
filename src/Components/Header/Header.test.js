import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';


import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe('Header', () => {
  it('should match the snapshot', () => {

    const mockMovie = {
      title: 'StarWars',
    }
    
    const mockError1 = null;

    const wrapper = shallow(<Header 
      movie={mockMovie}
      favoritesCount={(2)}
      error={mockError1}
    />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should show an error if an error is thrown', () => {

    const mockMovie = {
      title: 'StarWars',
    }
    
    const mockError2 = 'There was an error.'

    const wrapper = shallow(<Header 
      movie={mockMovie}
      favoritesCount={(2)}
      error={mockError2}
    />)

    const expected = wrapper.find('p').props().children;

    expect(expected).toEqual(mockError2)

  })

  it('should show some text if no movie nor error yet', () => {

    const mockMovie = null;
    
    const mockError3 = null;

    const wrapper = shallow(<Header 
      movie={mockMovie}
      favoritesCount={(2)}
      error={mockError3}
    />)

    const expected = wrapper.find('h4').props().children;

    expect(expected).toEqual('A long time ago in a galaxy far, far away ...')

  })
})