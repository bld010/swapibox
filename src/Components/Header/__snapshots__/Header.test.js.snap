// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="Header"
>
  <h1>
    SWAPIbox
  </h1>
  <TextCrawl
    movie={
      Object {
        "title": "StarWars",
      }
    }
  />
  <nav>
    <NavLink
      to="/people"
    >
      people
    </NavLink>
    <NavLink
      to="/planets"
    >
      planets
    </NavLink>
    <NavLink
      to="/vehicles"
    >
      vehicles
    </NavLink>
    <NavLink
      to="/favorites"
    >
      favorites | 
      2
    </NavLink>
  </nav>
</header>
`;
